server/apps/users/views.py

```diff
@@ -52,75 +63,65 @@ def post(self, request):
         response = Response()
         response.set_cookie(key='jwt', value=access_token, httponly=True)
 
-        # Generate JWT refresh token
-        refresh_token = generate_jwt_refresh_token(user)
-        response.set_cookie(key='refresh_token', value=refresh_token, httponly=True)
-
         response.data = {
             'message': AUTHENTICATION_SUCCESSFUL
         }
 
         return response
 
 
-class RefreshTokenView(APIView):
-    # Using @csrf_protect decorator for making sure that the cookie (refresh token) is not compromised
-    csrf_protect_method = method_decorator(csrf_protect)
-
-    @csrf_protect_method
-    def post(self, request):
-        refresh_token = request.COOKIES.get('refresh_token')
-
-        payload = verify_jwt_refresh_token(refresh_token)
+class UserView(generics.RetrieveAPIView):
+    serializer_class = UserSerializer
+    permission_classes = [IsAuthenticated]
+    authentication_classes = [JSONWebTokenAuthentication]
 
-        user = User.objects.filter(id=payload['id']).first()
-
-        # Create JWT access token
-        access_token = generate_jwt_access_token(user)
-
-        # Set JWT token as cookie. Set it as HTTP only so that no frontend can access the JWT token
-        response = Response()
-        response.set_cookie(key='jwt', value=access_token, httponly=True)
-
-        response.data = {
-            'message': ACCESS_TOKEN_GENERATED
-        }
-
-        return response
-
-
-class UserView(APIView):
     @staticmethod
-    def get(request):
-        token = request.COOKIES.get('jwt')
-
-        payload = verify_jwt_token(token)
-
-        user = User.objects.filter(id=payload['id']).first()
+    def get(request, **kwargs):
+        user = request.user
         serializer = UserSerializer(user)
 
         return Response(serializer.data)
 
 
 class LogoutView(APIView):
+    permission_classes = [IsAuthenticated]
+
     @staticmethod
     def post(request):
         response = Response()
+        blacklist_jwt_token(request.COOKIES.get('jwt'))
         response.delete_cookie('jwt')
+        response.delete_cookie('csrftoken')
         response.data = {
             'message': LOGOUT_SUCCESSFUL
         }
         return response
 
 
-class TodoListView(APIView):
+class TodoView(generics.CreateAPIView, generics.RetrieveUpdateDestroyAPIView):
+    permission_classes = [IsAuthenticated]
+    authentication_classes = [JSONWebTokenAuthentication]
+
+    # Create to'do
     @staticmethod
-    def get(request):
-        token = request.COOKIES.get('jwt')
+    def post(request, **kwargs):
+        user = request.user
+
+        # Set current user as a owner and creator of the to'do
+        request.data['owner'] = user.id
+        request.data['created_by'] = user.name
```

Comment:
check about request context in serializers context.
This type of thing you can do in the `save` method of the serializer or just override the `perform_create` and do it in that method..

Reply:
Updated in latest commit